pkg/mimirtool/commands/analyse_grafana.go

```diff
@@ -85,7 +82,10 @@ func AnalyzeGrafana(ctx context.Context, c *sdk.Client, folders []string) (*anal
 		if filterOnFolders && !slices.Contains(folders, link.FolderTitle) {
 			continue
 		}
-		data, _, err := c.GetRawDashboardByUID(ctx, link.UID)
+		fetchCtx, cancel := context.WithTimeout(context.Background(), readTimeout)
+		defer cancel()
```

Comment:
This `cancel` will only be executed once `AnalyzeGrafana` returns, but I think we want to cancel the context before the next iteration of the loop. Perhaps we could move the body of the loop to its own function?

Reply:
Hi @charleskorn, thanks for the feedback! I've made the changes as suggested by creating child contexts and moving the loop body to analyzeDashboard. Your input is greatly appreciated.